src/requestVariableCacheValueProcessor.ts

```diff
@@ -1,114 +1,83 @@
-                    }
-                }
-                const [, ...bodyParts]  = responseParts;
-                return RequestVariableCacheValueProcessor.resolveParts(
-                    bodyValue,
-                    bodyParts
-                );
-            } else {
-                return RequestVariableCacheValueProcessor.resolveParts(
-                    response,
-                    responseParts
-                );
-            }
-        } else {
-            return response;
-        }
-    }
+    public static resolveHttpPart(http: HttpRequest | HttpResponse, httpPart: string, rest?: string): any {
+        if (httpPart === "body") {
+            const { body } = http;
 
-    private static resolveParts(value: any, parts: string[]): any {
-        for (let i = 0; i < parts.length; i++) {
-            let part = parts[i];
-            let matches = part.match(partRegex);
-            const partName = matches[1];
-            value = value[partName];
-            // If array indices:
-            for (let j = 2; j < matches.length; j++) {
-                if (matches[j]) {
-                    value = value[matches[j].match(arrayIndexRegex)[1]];
-                }
+            if (!body) {
+                return undefined;
             }
-        }
 
-        return value;
-    }
+            const parsedBody = JSON.parse(body as string);
 
-    public static isValidPath(path: string): boolean {
-        return validPathRegex.test(path);
-    }
+            if (!rest) {
+                return parsedBody;
+            }
 
-    public static parseResponseBody(response: HttpResponse) {
-        const {body} = response;
-        const contentType = response.getResponseHeaderValue("content-type");
-        if (contentType) {
-            let mime = MimeUtility.parse(contentType);
-            let type = mime.type;
-            let suffix = mime.suffix;
-            if (type === 'application/json' || suffix === '+json') {
-                try {
-                    return JSON.parse(body);
-                } catch {
-                    window.showWarningMessage('The content type of response is application/json, while response body is not a valid json string');
-                }
-            } else {
-                window.showWarningMessage('Only JSON response is supported.');
+            return parsedBody ? RequestVariableCacheValueProcessor.resolveHttpBody(parsedBody, rest) : undefined;
+        } else if (httpPart === "headers") {
+            const { headers } = http;
+            if (!rest) {
+                return headers;
             }
+
+            return headers ? headers[rest] : undefined;
+        } else {
+            return undefined;
         }
+    }
 
-        return body;
+    public static resolveHttpBody(body: any, path: string) {
+        try {
+            const result = jp.query(body, path);
+            return result ? result[0] : null;
```

Comment:
I think we should check the type of result[0] here, if the result is `string`, just simply return, otherwise, `JSON.stringify(value)` it. The difference here is that `stringify` a raw string is not equivalent to its original one.  
![image](https://user-images.githubusercontent.com/2796177/36584507-ed8c0330-18b4-11e8-86c4-adcc0312e665.png)

Reply:
Good point, I'll update this

Reply:
I instead moved stringified outside of `requestVariableCacheValueProcessor`, because `requestVariableCompletionItemProvider` needs the actual (non-string) value to provide completion items of objects.